lib/node_modules/@stdlib/array/base/to-inserted-at/lib/assign.js

```diff
@@ -0,0 +1,236 @@
+	}
+	out[ offset + (index * stride) ] = value;
+
+	return out;
+}
+
+/**
+* Copies elements from one accessor array to another accessor array and inserts the element at the specified index to a provided value.
+*
+* @private
+* @param {Object} x - input array object
+* @param {integer} index - element index
+* @param {*} value - inserting value
+* @param {Object} out - output array object
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array
+*
+* @example
+* var toAccessorArray = require( '@stdlib/array/base/to-accessor-array' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = toAccessorArray( [ 1, 2, 3, 4 ] );
+*
+* var out = toAccessorArray( [ 0, 0, 0, 0, 0 ] );
+* var arr = accessors( arraylike2object( x ), 0, 5, arraylike2object( out ), 1, 0 );
+*
+* var v = arr.get( 0 );
+* // returns 5
+*/
+function accessors( x, index, value, out, stride, offset ) {
+	var xdata;
+	var odata;
+	var xget;
+	var oget;
+	var oset;
+	var io;
+	var i;
+
+	xdata = x.data;
+	odata = out.data;
+
+	xget = x.accessors[ 0 ];
+	oget = out.accessors[ 0 ];
+	oset = out.accessors[ 1 ];
+
+	io = offset;
+	for ( i = 0; i < xdata.length; i++ ) {
+		oset( odata, io, xget( xdata, i ) );
+		io += stride;
+	}
+	for ( i = xdata.length - 1; i >= index; i-- ) {
+		oset( odata, offset + ((i + 1) * stride), oget( odata, offset + (i * stride) ) );
+	}
+
+	oset( odata, offset + (index * stride), value );
+	return odata;
+}
+
+/**
+* Copies elements from one complex array to another complex array and inserts the element at the specified index to a provided value.
+*
+* @private
+* @param {Collection} x - real-valued floating-point input array view
+* @param {integer} index - element index
+* @param {ComplexLike} value - inserting value
+* @param {Collection} out - real-valued floating-point output array view
+* @param {integer} stride - output array stride
+* @param {NonNegativeInteger} offset - output array offset
+* @returns {Collection} output array view
+*
+* @example
+* var Complex128 = require( '@stdlib/complex/float64/ctor' );
+* var Float64Array = require( '@stdlib/array/float64' );
+* var arraylike2object = require( '@stdlib/array/base/arraylike2object' );
+*
+* var x = new Float64Array( [ 1.0, 2.0, 3.0, 4.0 ] );
+*
+* var out = new Float64Array( 6 );
+* var arr = complex( x, 0, new Complex128( 5.0, 6.0 ), out, 1, 0 );
+* // returns <Float64Array>[ 5.0, 6.0, 1.0, 2.0, 3.0, 4.0 ]
+*/
+function complex( x, index, value, out, stride, offset ) {
+	var so;
+	var io;
+	var i;
+
+	so = stride * 2; // multiply by 2, as real-valued array consists of interleaved real and imaginary components
+	io = offset * 2;
+	for ( i = 0; i < x.length/2; i++ ) {
+		out[ io + ( i * so ) ] = x[ 2 * i ];
+		out[ io + ( i * so ) + 1 ] = x[ ( 2 * i ) + 1 ];
+	}
+	for ( i = ( x.length/2 ) - 1; i >= index; i-- ) {
+		out[ io + ( ( i + 1 ) * so ) ] = out[ io + ( i * so ) ];
```

Comment:
Also, don't inline indexing expressions like this. You are currently performing the same arithmetic 4 separate times: io + (i+1)*so. Simplify your expressions.